DigitalLearningSolutions.Data/Models/User/UserCentreAccounts.cs

```diff
@@ -0,0 +1,35 @@
+﻿namespace DigitalLearningSolutions.Data.Models.User
+{
+    public class UserCentreAccounts
+    {
+        public UserCentreAccounts(
+            int centreId,
+            AdminAccount? adminAccount = null,
+            DelegateAccount? delegateAccount = null
+        )
+        {
+            CentreId = centreId;
+            AdminAccount = adminAccount;
+            DelegateAccount = delegateAccount;
+        }
+
+        public int CentreId { get; }
+        public AdminAccount? AdminAccount { get; }
+        public DelegateAccount? DelegateAccount { get; }
+        public bool IsCentreActive => AdminAccount?.CentreActive ?? DelegateAccount?.CentreActive ?? false;
+
+        public string? CentreName => AdminAccount?.CentreName ?? DelegateAccount?.CentreName;
+        public AdminAccount? ActiveAdminAccount => AdminAccount?.Active == true ? AdminAccount : null;
+
+        public DelegateAccount? ActiveApprovedDelegateAccount =>
+            DelegateAccount is { Active: true, Approved: true } ? DelegateAccount : null;
+
+        // True if there is either an active admin account, or an active approved delegate account

```

Comment:
I wanted to be explicit about what the difference between `CanLogInToCentre` and `CanLogDirectlyInToCentre` is, because I don't think it's immediately clear that `CanLogDirectlyInToCentre` *is not a subset of* `CanLogInToCentre`. However, I agree that in their current form, these comments do not really add value. It would be better to explain the business decision as to why you will be taken to the choose a centre page when you have, for instance, an active delegate and an inactive admin account.

I think that particular example (active delegate and an inactive admin account) is an important one to explain, since, if that is your only centre, you will be taken to the choose a centre page and shown one row, with a "Delegate" tag, an "Active" tag, and a "Choose" button, and might rightly wonder why you are being forced into a situation where you need to click this button despite having no other choices.